vscode/src/completions/logger.ts

```diff
@@ -512,10 +523,43 @@ export function loaded(
     if (!event.loadedAt) {
         event.loadedAt = performance.now()
     }
-
     if (event.items.length === 0) {
         event.items = items.map(item => completionItemToItemInfo(item, isDotComUser))
     }
+
+    // 🚨 SECURITY: included only for DotCom users & Public github Repos.
+
+    if (
+        isDotComUser &&
+        inlineContextParams?.gitUrl &&
+        event.params.inlineCompletionItemContext === undefined
+    ) {
+        const instance = RepoMetadatafromGitApi.getInstance()
+        const gitRepoMetadata = instance.getRepoMetadataIfCached(inlineContextParams.gitUrl)
+        if (gitRepoMetadata === undefined || gitRepoMetadata.isPublic === false) {
+            return
+        }
+        const contextSnippets: InlineCompletionItemRetrivedContext[] = []
+        if (inlineContextParams?.context) {
+            for (const snippet of inlineContextParams.context) {
+                contextSnippets.push({
+                    content: snippet.content,
+                    startLine: snippet.startLine,
+                    endLine: snippet.endLine,
+                    filePath: snippet.uri.fsPath,
+                })
+            }
+        }
+        event.params.inlineCompletionItemContext = {
+            gitUrl: inlineContextParams.gitUrl,
+            commit: inlineContextParams.commit,
+            prefix: params.docContext.prefix,
+            suffix: params.docContext.suffix,
+            triggerLine: params.position.line,
+            triggerCharacter: params.position.character,
+            context: contextSnippets,
+        }
```

Comment:
nit: concise version with the same level of type-safety:

```suggestion
        event.params.inlineCompletionItemContext = {
            gitUrl: inlineContextParams.gitUrl,
            commit: inlineContextParams.commit,
            prefix: params.docContext.prefix,
            suffix: params.docContext.suffix,
            triggerLine: params.position.line,
            triggerCharacter: params.position.character,
            context: inlineContextParams.context.map(snippet => ({
                    content: snippet.content,
                    startLine: snippet.startLine,
                    endLine: snippet.endLine,
                    filePath: snippet.uri.fsPath,
            })),
        }
```